File: src/main/java/com/efedotov/vaultly/dto/file/FileUploadRequest.java
package com.efedotov.vaultly.dto.file;

import org.springframework.web.multipart.MultipartFile;

import lombok.Data;

@Data
public class FileUploadRequest {
    private MultipartFile file;
    private Long folderId;
    private Boolean encrypt = false;
    private String encryptionPassword;
}

File: lib/server/model/file/file_upload_request/file_upload_request.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:http/http.dart' as http;

part 'file_upload_request.freezed.dart';

@freezed
abstract class FileUploadRequest with _$FileUploadRequest {
  const factory FileUploadRequest({
    required http.MultipartFile file,
    int? folderId,
    bool? encrypt,
    String? encryptionPassword,
  }) = _FileUploadRequest;
}
